Core/include/Acts/Utilities/RangeXD.hpp

```diff
@@ -0,0 +1,228 @@
+
+#include "Acts/Utilities/Range1D.hpp"
+
+#include <algorithm>
+#include <limits>
+
+namespace Acts {
+/// @brief An orthogonal range in an arbitrary number of dimensions
+///
+/// By combining a number one-dimensional ranges we can (under the assumption
+/// that our axes are orthogonal) construct an orthogonal range of values. In
+/// other words, a hyperrectangular volume in space.
+///
+/// @tparam Dims The number of dimensions in our range
+/// @tparam Type The scalar type of our ranges
+/// @tparam Vector The vector type used to define coordinates
+template <std::size_t Dims, typename Type,
+          template <typename, std::size_t> typename Vector = std::array>
+class RangeXD {
+ public:
+  /// @brief The type used to describe coordinates in our range
+  using coordinate_t = Vector<Type, Dims>;
+
+  /// @brief Determine whether this range is degenerate
+  ///
+  /// A degenerate multi-dimensional range has no volume and cannot contain any
+  /// values. This is the case if any of its dimensions are degenerate.
+  ///
+  /// @return true The range is degenerate
+  /// @return false The range is not degenerate
+  bool degenerate(void) const {
+    for (std::size_t i = 0; i < Dims; ++i) {
+      if (m_dims[i].degenerate()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /// @brief Determine whether the range contains a certain point
+  ///
+  /// This is true if and only if the range contains the point in all of its
+  /// dimensions.
+  ///
+  /// @param v The coordinate to check for membership in the range
+  ///
+  /// @return true The coordinate is inside the range
+  /// @return false The coordinate is outside the range
+  bool contains(const coordinate_t& v) const {
+    for (std::size_t i = 0; i < Dims; ++i) {
+      if (!m_dims[i].contains(v[i])) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /// @brief Access one of the dimensional ranges of the volume
+  ///
+  /// @param i The index of the dimension to access
+  /// @return A reference to the dimension contained in this range
+  Range1D<Type>& operator[](const std::size_t& i) {
+    return m_dims[static_cast<std::size_t>(i)];
+  }
+
+  /// @brief Access one of the dimensional ranges of the volume
+  ///
+  /// @param i The index of the dimension to access
+  /// @return A reference to the dimension contained in this range
+  const Range1D<Type>& operator[](const std::size_t& i) const {
+    return m_dims[static_cast<std::size_t>(i)];
+  }
+
+  /// @brief Determine whether two ranges are equal
+  ///
+  /// Two n-dimensional ranges are equal if and only if they are equal in each
+  /// of their n dimensions.
+  ///
+  /// @param o The other range to check for equality
+  ///
+  /// @return true The ranges are equal
+  /// @return false The ranges are not equal
+  bool operator==(const RangeXD<Dims, Type, Vector>& o) const {
+    for (std::size_t i = 0; i < Dims; ++i) {
+      if (!(this->m_dims[i] == o[i])) {
+        return false;
```

Comment:
I suppose you could rely on the fact that the `Vector` typename implements an `==` operator, but also ok this way.

Reply:
A sorry, my mistake - you rely on this already, this concerns the dimensions.